azure-kusto-data/azure/kusto/data/client.py

```diff
@@ -1,962 +1,975 @@
+            if properties.client_request_id is not None:
+                request_headers["x-ms-client-request-id"] = properties.client_request_id
+            if properties.application is not None:
+                request_headers["x-ms-app"] = properties.application
+            if properties.user is not None:
+                request_headers["x-ms-user"] = properties.user
+            if properties.get_option(ClientRequestProperties.no_request_timeout_option_name, False):
+                timeout = KustoClient._mgmt_default_timeout
+            else:
+                timeout = properties.get_option(ClientRequestProperties.request_timeout_option_name, timeout)
+
+        timeout = (timeout or KustoClient._mgmt_default_timeout) + KustoClient._client_server_delta
+
+        self.json_payload = json_payload
+        self.request_headers = request_headers
+        self.timeout = timeout
+
+
+class HTTPAdapterWithSocketOptions(requests.adapters.HTTPAdapter):
+    def __init__(self, *args, **kwargs):
+        self.socket_options = kwargs.pop("socket_options", None)
+        self.pool_maxsize = kwargs.pop("pool_maxsize", None)
+        self.max_retries = kwargs.pop("max_retries", None)
+        super(HTTPAdapterWithSocketOptions, self).__init__(*args, **kwargs)
+
+    def init_poolmanager(self, *args, **kwargs):
+        if self.socket_options is not None:
+            kwargs["socket_options"] = self.socket_options
+        super(HTTPAdapterWithSocketOptions, self).init_poolmanager(*args, **kwargs)
+
+
+class _KustoClientBase(abc.ABC):
+    API_VERSION = "2019-02-13"
+
+    _mgmt_default_timeout = timedelta(hours=1, seconds=30)
+    _query_default_timeout = timedelta(minutes=4, seconds=30)
+    _streaming_ingest_default_timeout = timedelta(minutes=10)
+
+    _auth_provider: _AadHelper
+
+    def __init__(self, kcsb: Union[KustoConnectionStringBuilder, str]):
+        self._kcsb = kcsb
+        self._proxy: Optional[str] = None
+        self._proxy_dict: Optional[Dict[str, str]] = None
+        if not isinstance(kcsb, KustoConnectionStringBuilder):
+            self._kcsb = KustoConnectionStringBuilder(kcsb)
+        self._kusto_cluster = self._kcsb.data_source
+
+        # Create a session object for connection pooling
+        self._mgmt_endpoint = "{0}/v1/rest/mgmt".format(self._kusto_cluster)
+        self._query_endpoint = "{0}/v2/rest/query".format(self._kusto_cluster)
+        self._streaming_ingest_endpoint = "{0}/v1/rest/ingest/".format(self._kusto_cluster)
+        self._request_headers = {
+            "Accept": "application/json",
+            "Accept-Encoding": "gzip,deflate",
+            "x-ms-client-version": "Kusto.Python.Client:" + VERSION,
+            "x-ms-version": self.API_VERSION,
+        }
+
+    def set_proxy(self, proxy: str):
+        self._proxy = proxy
+        self._proxy_dict = {"http": proxy, "https": proxy}
+        self._auth_provider.token_provider.set_proxy_dict(self._proxy_dict)
```

Comment:
set_proxy_dict with no args defaults proxy_dict to "{"http": proxy, "https": proxy}"

Reply:
And remove proxy_dict from this (and any similar) classes